core/shared/src/main/scala/zio/prelude/data/Optional.scala

```diff
@@ -13,7 +14,7 @@ import scala.language.implicitConversions
  * The only difference between this type and [[scala.Option]] is that there is an implicit
  * conversion defined from `A`` to `Optional[A]`, and from `Option[A]`` to `Optional[A]`.
  */
-sealed trait Optional[+A] { self =>
+sealed trait Optional[+A] extends IterableOnce[A] { self =>
```

Comment:
Why not `Iterable`?

Reply:
Fair enough, but the question still stands. Wouldn't it be better to extend `Iterable` since we're changing things anyway? Maybe there are other use cases which won't do with mere `IterableOnce`. Or do you think it will be more complicated?

Reply:
> Why not Iterable?

Because `Option` doesn't extend `Iterable` but `IterableOnce` 🤷‍♂️
I'm just copying `Option` code

Reply:
Ah, I see. Than that should be good enough for us too :)